File: src/NocWMS-Server/Server.ApiService/Models/OutboundReceipt.cs
using System;
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;

namespace Server.ApiService.Models
{
    public class OutboundReceipt : IEntity
    {
        public string Id { get; set; }
        public int WarehouseId { get; set; }
        public string CustomerId { get; set; }
        public string WaybillNo { get; set; }
        public float Total { get; set; }
        public string StaffId { get; set; }
        public string HandlerName { get; set; }
        public string Acceptor { get; set; }
        public string Deliveryman { get; set; }
        public bool IsDone { get; set; }
        public DateTime CreateTime { get; set; }

        public Warehouse Warehouse { get; set; }
        public Staff Staff { get; set; }
        public Customer Customer { get; set; }
        public List<OutboundReceiptItem> OutboundReceiptItems { get; set; }
    }
}

File: src/NocWMS-Client/src/app/models/outbound-receipt.ts
import { OutboundReceiptItem } from "./outbound-receipt-item";

export class OutboundReceipt {
    constructor(outboundReceipt?: Partial<OutboundReceipt>) {
        if (outboundReceipt) {
            this.id = outboundReceipt.id;
            this.staffId = outboundReceipt.staffId;
            this.warehouseId = outboundReceipt.warehouseId;
            this.warehouseName = outboundReceipt.warehouseName;
            this.customerId = outboundReceipt.customerId;
            this.customerName = outboundReceipt.customerName;
            this.handlerName = outboundReceipt.handlerName;
            this.acceptor = outboundReceipt.acceptor;
            this.deliveryman = outboundReceipt.deliveryman
            this.waybillNo = outboundReceipt.waybillNo;
            this.isDone = outboundReceipt.isDone;
            this.createTime = outboundReceipt.createTime;
            this.outboundReceiptItems = outboundReceipt.outboundReceiptItems;
        }
    }

    public id: string;
    public staffId: string;
    public warehouseId: number;
    public warehouseName: string;
    public customerId: string;
    public customerName: string;
    public handlerName: string;
    public acceptor: string;
    public deliveryman: string;
    public waybillNo: string;
    public isDone: boolean;
    public createTime: string;
    public outboundReceiptItems: OutboundReceiptItem[];
    public get total(): number {
        let sum = 0;
        if (this.outboundReceiptItems) {
            this.outboundReceiptItems.forEach((v) => sum += v.productPrice * v.num);
        }

        return sum;
    }
}